src/main/java/com/google/jenkins/plugins/computeengine/InstanceConfiguration.java

```diff
@@ -234,7 +275,33 @@ public ComputeEngineInstance provision(TaskListener listener, Label requiredLabe
             Instance i = instance();
             Operation operation = cloud.client.insertInstance(cloud.projectId, i);
             logger.println("Sent insert request");
-            ComputeEngineInstance instance = new ComputeEngineInstance(cloud.name, i.getName(), i.getZone(), i.getDescription(), runAsUser, "./.jenkins-slave", numExecutors, mode, requiredLabel == null ? "" : requiredLabel.getName(), new ComputeEngineLinuxLauncher(cloud.getCloudName(), operation, this.useInternalAddress), new CloudRetentionStrategy(retentionTimeMinutes), getLaunchTimeoutMillis());
+            String targetRemoteFs = this.remoteFs;
+            ComputeEngineComputerLauncher launcher = null;
+            if (this.windows) {
+                launcher = new ComputeEngineWindowsLauncher(cloud.getCloudName(), operation, this.useInternalAddress);
+                if (targetRemoteFs == null || targetRemoteFs.isEmpty()) {
```

Comment:
Same comment as above for the target remote FS. Also for better orthogonality, you should separate the logic which maps to the target remote FS.